buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/ci/metrics/DrainingBuildListener.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins.ci.metrics;
+
+import org.gradle.BuildAdapter;
+import org.gradle.BuildResult;
+import org.gradle.api.logging.Logger;
+
+class DrainingBuildListener extends BuildAdapter {
+  private final long sleepDuration;
+  private final Logger logger;
+
+  DrainingBuildListener(long sleepDuration, Logger logger) {
+    this.sleepDuration = sleepDuration;
+    this.logger = logger;
+  }
+
+  @Override
+  public void buildFinished(BuildResult result) {
+    try {
+      logger.warn("Draining metrics to Stackdriver.");
+      Thread.sleep(sleepDuration);
```

Comment:
What is the purpose of this?

Reply:
added some comments.

As for the upload period, I did not see that(thanks) but looking now, the doc mentions:

> If you are adding data points to different time series, then there is no rate limitation.

So we should be fine as we don't write to the same timeseries more than once per gradle run, I will double check that though and let you know.

Reply:
@allisonbm92 So I can confirm that the period is not a problem for us as we never write the same timeseries more than once per ci run